pkg/app/piped/cloudprovider/ecs/client.go

```diff
@@ -145,6 +155,38 @@ func (c *client) CreateTaskSet(ctx context.Context, service types.Service, taskD
 	return output.TaskSet, nil
 }
 
+func (c *client) GetPrimaryTaskSet(ctx context.Context, service types.Service) (*types.TaskSet, error) {
+	input := &ecs.DescribeServicesInput{
+		Cluster: service.ClusterArn,
+		Services: []string{
+			*service.ServiceArn,
+		},
+	}
+	output, err := c.client.DescribeServices(ctx, input)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get primary task set of service %s: %w", *service.ServiceName, err)
+	}
+	taskSets := output.Services[0].TaskSets
+	for _, taskSet := range taskSets {
+		if aws.ToString(taskSet.Status) == "PRIMARY" {
+			return &taskSet, nil
+		}
+	}
+	return nil, nil
```

Comment:
Not found should be an error as we always do.

Reply:
Addressed on https://github.com/pipe-cd/pipe/pull/2007/commits/8808feb81ca5eafb6b84b585fb4701a0dbba5d9b 🙏